lib/util/string-functions.js

```diff
@@ -4,13 +4,29 @@ const WHITESPACE_CHAR_CODE = 32
 const TAB_CHAR_CODE = 9
 const DOT_CHAR_CODE = 46
 
+exports.removeLeadingWhitespaces = function removeLeadingWhitespaces (string) {
+  while (string.charCodeAt(0) === WHITESPACE_CHAR_CODE) {
+    string = string.substring(1, string.length)
+  }
+  return string
+}
+
+exports.removeTrailingWhitespaces = function removeTrailingWhitespaces (string) {
+  while (string.charCodeAt(string.length - 1) === WHITESPACE_CHAR_CODE) {
+    string = string.substring(0, string.length - 1)
+  }
+  return string
+}
+
 exports.isDigit = function isDigit (charCode) {
   return charCode >= MIN_DIGIT_CHAR_CODE && charCode <= MAX_DIGIT_CHAR_CODE
 }
 
 exports.isNumber = function isNumber (string) {
-  for (var i = 0; i < string.length; i++) {
-    const charCode = string.charCodeAt(i)
+  let hasNoWhiteSpace = exports.removeLeadingWhitespaces(string)
+  hasNoWhiteSpace = exports.removeTrailingWhitespaces(hasNoWhiteSpace)
```

Comment:
Are you sure this does not break functionality elsewhere? This is also inconsistent with how the other exported functions work since they don't trim whitespace

Reply:
I can't be sure. This function wasn't covered in the test case. In that case, I will change it back and create another function.

Reply:
why not just have whoever needs the whitespace removal actually call the relevant functions?

Reply:
That will work too. I'll do that.